docs/DeveloperGuide.md

```diff
@@ -1,34 +1,726 @@
 # Developer Guide
 
-## Design & implementation
+## Introduction
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+`FridgeFriend` is an app for managing food in the fridge, optimised for use via a Command Line Interface (CLI).
+If you can type fast, `FridgeFriend` can track your cold or frozen groceries faster and easier than any other apps.
+It is written in Java, and has more than 4.3kLoC.
 
+## Contents
 
-## Product scope
-### Target user profile
+* [Introduction](#introduction)
+* [Design](#design)
+  * [Architecture](#architecture)
+  * [Utilities Component](#utilities-component)
+  * [Command Component](#command-component)
+  * [Food Component](#food-component)
+  * [Exception Component](#exception-component)
+* [Implementation](#implementation)
+  * [Main Logic](#main-logic)
+  * [Add Command](#add-command)
+  * [Remove Command](#remove-command)
+  * [Search Command](#search-command)
+  * [Running Low Command](#running-low-command)
+  * [Set Limit Command](#set-limit-command)
+  * [Expiring Command](#expiring-command)
+  * [List Command](#list-command)
+  * [History Command](#history-command)
+* [Product Scope](#product-scope)
+  * [Target User Profile](#target-user-profile)
+  * [Value Proposition](#value-proposition)
+* [User Stories](#user-stories)
+* [Non-Functional Requirements](#non-functional-requirements)
+* [Glossary](#glossary)
+* [Instructions for Manual Testing](#appendix-instructions-for-manual-testing)
+* [Attribution](#attribution)
 
-{Describe the target user profile}
+## Design
 
-### Value proposition
+This section focuses on the general architecture of the application as well as the relationships between the classes involved. The overall architecture would be covered first, before the four individual components.
 
-{Describe the value proposition: what problem does it solve?}
+### Architecture
+
+![Architecture Diagram](diagrams/diagram_images/ArchitectureDiagram.png)
+
+The ***Architecture Diagram*** given above explains the high-level design of the App.
+Given below is a quick overview of each component.
+
+The Main driver class for the FridgeFriend app is
+named **[`FridgeFriend`](https://github.com/AY2021S2-CS2113-T10-1/tp/blob/master/src/main/java/seedu/fridgefriend/FridgeFriend.java)**.
+It is responsible for,
+
+* At app launch: Initializes the components in the correct sequence, and connects them up with each other.
```

Comment:
Some parts are labelled as "app" and some parts are labelled as "App", maybe can standardize to "App" to make it more consistent?